### PR TITLE
Use error=true in Obsolete attribute on IsByRefLikeAttribute for ref struct

### PR DESCRIPTION
**Pull request by jcouv**
**Customer scenario**
Create a new ref-like type (new C# 7.2 feature) and refer to that type from a project that uses an older compiler (or the compiler for some other language). The ref-like type should be poisoned (produce an error) in those compilers, since they don't know how to honor the safety guarantees of such types.
But currently, the `Obsolete` attribute that we use for poisoning is only marked with error=false, which means a warning is produced on usage, rather than an error.

**Bugs this fixes:**
Fixes https://github.com/dotnet/roslyn/issues/22447

**Risk**
**Performance impact**
Low. We're just strengthening the poisoning by using error=true in `Obsolete`.

**Is this a regression from a previous update?**
No.

### CODE REVIEW

**Comment by jcouv** [comment]
@OmarTawfik @dotnet/roslyn-compiler for review (tiny change). Thanks

File: src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_IsByRefLike.cs [hunk 1]
```diff
@@ -871,6 +871,36 @@ public struct S1 {}
...
+                Assert.Equal(accessibility, attributeType.DeclaredAccessibility);
+
+                var attribute = attributes[1];
+                Assert.Equal("System.ObsoleteAttribute", attribute.AttributeClass.ToDisplayString());
+                TypedConstant[] constructorArguments = attribute.ConstructorArguments.ToArray();
+                Assert.Equal(2, constructorArguments.Length);
+                Assert.Equal("Types with embedded references are not supported in this version of your compiler.", constructorArguments[0].Value);
+                Assert.Equal(true, constructorArguments[1].Value);
+            });
+        }
```
Comment (cston) [inline src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_IsByRefLike.cs]: Should we verify an error is reported when using from VB?

File: src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol.cs [hunk 2]
```diff
@@ -1136,7 +1136,7 @@ internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, r
                     AddSynthesizedAttribute(ref attributes, compilation.TrySynthesizeAttribute(WellKnownMember.System_ObsoleteAttribute__ctor,
                         ImmutableArray.Create(
                                        new TypedConstant(compilation.GetSpecialType(SpecialType.System_String), TypedConstantKind.Primitive, PEModule.ByRefLikeMarker),
-                                       new TypedConstant(compilation.GetSpecialType(SpecialType.System_Boolean), TypedConstantKind.Primitive, false)), 
+                                       new TypedConstant(compilation.GetSpecialType(SpecialType.System_Boolean), TypedConstantKind.Primitive, true)),
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol.cs]: >new TypedConstant(compilation.GetSpecialType(SpecialType.System_Boolean), TypedConstantKind.Primitive, true)), [](start = 39, length = 110)

consider adding a comment describing the meaning of values.

File: src/Compilers/VisualBasic/Test/Emit/Attributes/AttributeTests_ObsoleteAttribute.vb [hunk 3]
```diff
@@ -1469,6 +1469,39 @@ BC40000: 'Public Sub Bar()' is obsolete: 'hi'.
...
+    </file>
+</compilation>
+
+            Dim vbCompilation = CreateCompilationWithMscorlibAndVBRuntimeAndReferences(vbSource, {ref})
+
+            vbCompilation.AssertTheseDiagnostics((<![CDATA[
+BC30668: 'S' is obsolete: 'Types with embedded references are not supported in this version of your compiler.'.
+    Sub M(s As S)
+               ~
+]]>))
```
Comment (cston) [inline src/Compilers/VisualBasic/Test/Emit/Attributes/AttributeTests_ObsoleteAttribute.vb]: `AssertTheseDiagnostics` should not be necessary. `VerifyDiagnostics` should be sufficient.

Comment (jcouv) [reply src/Compilers/VisualBasic/Test/Emit/Attributes/AttributeTests_ObsoleteAttribute.vb]: The reason I kept the original check is that it shows where the error occurs.

**Comment by jcouv** [comment]
@jaredpar for ask-mode approval. Thanks

**Comment by jcouv** [comment]
@jaredpar for ask-mode approval. Thanks

**Comment by jcouv** [comment]
@jaredpar for ask-mode approval. Thanks

**Comment by jaredpar** [comment]
approved for ask mode

**Pull request merged**